### PR TITLE
Mac docker

### PR DESCRIPTION
**Pull request by millerdev**
Make docker services work more smoothly on OS X with boot2docker

@biyeun I think you had some problems with Kafka and Docker recently. This should fix it. Also, the docker data home issue on Mac should be fixed now too.

@snopoke cc @dimagi/team-commcare-hq

### CODE REVIEW

File: docker/_include.sh [hunk 1]
```diff
@@ -1,25 +1,26 @@
 DOCKER_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-if [ `uname` == 'Linux' -o `uname` == 'Darwin' ]; then
+if [ `uname` == 'Linux' ]; then
```
Comment (nickpell) [inline docker/_include.sh]: 👍

File: docker/docker-services.sh [hunk 2]
```diff
@@ -16,14 +22,11 @@ function usage() {
...
 function runner() {
-    if [ `uname` == 'Linux' ]; then
-        sudo \
-            env ES_CLUSTER_NAME=$ES_CLUSTER_NAME DOCKER_DATA_HOME=$DOCKER_DATA_HOME \
-            docker-compose -f $DOCKER_DIR/compose/docker-compose-services.yml -p $PROJECT_NAME $@
-    else
-        env ES_CLUSTER_NAME=$ES_CLUSTER_NAME DOCKER_DATA_HOME=$DOCKER_DATA_HOME \
-            docker-compose -f $DOCKER_DIR/compose/docker-compose-services.yml -p $PROJECT_NAME $@
-    fi
+    $UDO \
```
Comment (millerdev) [inline docker/docker-services.sh]: set by `_include.sh`

File: docker/_include.sh (hunk 1 above)
Comment (biyeun) [inline docker/_include.sh]: YASSS

**Comment by biyeun** [comment]
THANK YOU 🎉 :+1: on ✅

File: docker/_include.sh (hunk 1 above)
Comment (benrudolph) [inline docker/_include.sh]: hurray! 🍎 🍏 📱

File: docker/utils.sh [hunk 3]
```diff
@@ -11,12 +17,12 @@ docker_run() {
 get_container_id() {
     label_name=$1
     value=$2
-    sudo docker ps -qf label=$label_name=$value -f status=running
+    $UDO docker ps -qf label=$label_name=$value -f status=running
```
Comment (benrudolph) [inline docker/utils.sh]: haha `$UDO`

File: docker/_include.sh [hunk 4]
```diff
@@ -1,25 +1,26 @@
...
     DOCKER_DATA_HOME=$XDG_DATA_HOME/dockerhq
+    UDO="sudo"
 else
     DOCKER_DATA_HOME=/data
+    UDO=""
 fi
 
-mkdir -p $DOCKER_DATA_HOME
+if [ `uname` == 'Darwin' ]; then
+    docker-machine ssh $DOCKER_MACHINE_NAME sudo mkdir -p $DOCKER_DATA_HOME
```
Comment (millerdev) [inline docker/_include.sh]: On Mac `DOCKER_DATA_HOME` is a directory inside the boot2docker VM, so be warned that if you delete your VM you'll lose all the data.

For performance reasons we probably do not want to put it in a shared folder that would locate it outside the VM on the host Mac filesystem. From what I've read that can be super slow. Maybe native Docker on Mac will make it possible?

**Pull request merged**